PROTO_tests/tests/groupby_test.py

```diff
@@ -0,0 +1,577 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+import arkouda as ak
+from arkouda.groupbyclass import GroupByReductionType
+
+
+def to_tuple_dict(labels, values):
+    # transforms labels from list of arrays into a list of tuples by index and builds a dictionary
+    # labels: [array(['b', 'a', 'c']), array(['b', 'a', 'c'])] -> [('b', 'b'), ('a', 'a'), ('c', 'c')]
+    return {
+        label: value
+        for (label, value) in zip(
+            [index_tuple for index_tuple in zip(*[pda.to_ndarray() for pda in labels])],
+            values.to_ndarray(),
+        )
+    }
```

Comment:
Never look at old code. Either of these are a cleaner way to do this than what I originally had
```python
return {
    label: value
    for (label, value) in zip(
        list(zip(*[pda.to_ndarray() for pda in labels])),
        values.to_ndarray(),
    )
}
```
or
```python
return dict(zip(list(zip(*[pda.to_ndarray() for pda in labels])), values.to_ndarray()))
```

Reply:
For what it is worth, I am not sure this is 100% a necessary change as it will ultimately provide the same output. However, I agree its a bit cleaner, so its updated.